File: solutions/r/game-of-life/1/game-of-life.R
tick <- function(cells) {
  r <- nrow(cells)
  c <- ncol(cells)
  
  # Handle empty or 0-dimension matrices
  if (is.null(r) || is.null(c) || r == 0 || c == 0) {
    return(cells)
  }
  
  next_gen <- matrix(0, nrow = r, ncol = c)
  
  for (i in seq_len(r)) {
    for (j in seq_len(c)) {
      # Determine neighbor boundaries (clamped to matrix limits)
      row_indices <- max(1, i - 1):min(r, i + 1)
      col_indices <- max(1, j - 1):min(c, j + 1)
      
      # Sum the submatrix and subtract the cell's own value to get neighbor count
      neighbors <- sum(cells[row_indices, col_indices]) - cells[i, j]
      
      # Apply Conway's Game of Life rules
      if (cells[i, j] == 1) {
        if (neighbors == 2 || neighbors == 3) {
          next_gen[i, j] <- 1
        } else {
          next_gen[i, j] <- 0
        }
      } else {
        if (neighbors == 3) {
          next_gen[i, j] <- 1
        } else {
          next_gen[i, j] <- 0
        }
      }
    }
  }
  
  return(next_gen)
}

File: solutions/euphoria/game-of-life/1/game-of-life.ex
public function tick(sequence matrix)
  sequence res = matrix
  for i = 1 to length(matrix) do
    for j = 1 to length(matrix[i]) do
      res[i][j] = 0
      atom neighbors = 0
      for k = -1 to 1 do
        for l = -1 to 1 do
          if (k != 0 or l != 0) and i + k >= 1 and i + k <= length(matrix) and j + l >= 1 and j + l <= length(matrix[i]) then
            neighbors += matrix[i + k][j + l]
          end if
        end for
      end for
      if neighbors = 3 or (matrix[i][j] = 1 and neighbors = 2) then
        res[i][j] = 1
      end if
    end for
  end for
  return res
end function